unidist/core/backends/mpi/core/common.py

```diff
@@ -322,3 +424,25 @@ def _unwrap_data_id(*args):
     else:
         unwrapped = _unwrap_data_id(data_ids)
         return unwrapped, is_pending
+
+
+def is_shared_memory_supported():
+    """
+    Check if the unidist on MPI supports shared memory.
+
+    Returns
+    -------
+    bool
+        True or False.
+
+    Notes
+    -----
+    Prior to the MPI 3.0 standard there is no support for shared memory.
+    """
+    if not MpiUseSharedMemory.get():
+        return False
+
+    if MPI.VERSION < 3:
```

Comment:
We should also add a check for mpich versions that do not support the shared memory feature for c/w mode.